cookies implementaion

prabhamanju opened the pull request:
plz check for cookies....and let me know how i can improve.

prabhamanju committed:
cookies implementaion

archived-m commented on server/src/controller/user.controller.js:
```diff
@@ -130,8 +133,20 @@ exports.querylogin = (req, res) => {
               success: true,
               _id: user._id,
-              token: "manju" + token
+              token: "%$%#&$" + token
```
As discussed during class, adding your own complexity to a token hash will result in a no-op, and is bad practice.
Because you don't store this "unique string" in a config or env file, this complexity also loses it value, since your repository is open source. Anyone can see you add `%$%#&$` to your token.

prabhamanju replied:
thanks for reminding me !! I am using unique string in a .env file, I will remove this (token: "%$%#&$" + token)  from final code.

archived-m commented on server/src/controller/user.controller.js:
```diff
@@ -130,8 +133,20 @@ exports.querylogin = (req, res) => {
             res.json({
```
The reason your cookie isn't being set in the browser is because you're sending the response before adding the cookie header.

```
res
.cookie("token", token, {
  maxAge: 1000 * 60 * 10,
  httpOnly: true
})
.json({
  success: true,
  _id: user._id,
  token: token
});
```

You can restructure your code to first set a cookie header, then send the response. That should work.

Right now your token will still be `undefined`, which is your next problem to solve :)

archived-m replied:
Also note there is no reason to include the token in the response json body - it defeats the purpose of having an `httpOnly` cookie. Your token is in the cookie.

prabhamanju replied:
thanks a lot Michiel for clarification, I will try and update the code.

prabhamanju committed:
updated cookies
updated UserFormRegistrationBody
updated Header
updated user.controller for new registration
created function for new registration
some minor updates

prabhamanju committed:
minor update in menter.jsx
implemented some validations on registration form in userFormRegistration.jsx
minor update in controller.js
minor update in users.js
minor change in loginPageBody.jsx
